cpp/src/bitmask/bit_mask.cuh

```diff
@@ -153,20 +148,20 @@ inline gdf_error create_bit_mask(bit_mask_t **mask,
  *
  *  Note that for performance reasons (this is often called in inner loops
  *  in CUDA device code) this function does not verify that @p valid is non-null.
- *  That should be checked in a wider scope, since it usually doesn't vary per
- *  thread.
+ *  That should be checked in a wider scope, since it usually doesn't vary for 
+ *  different threads of the same kernel.
  *
  *  @param[in]  valid         The bit mask to update
- *  @param[in]  record_idx    The record index
+ *  @param[in]  bit_index     Index of the bit within the mask
  *
  *  @return which bit within the bit mask
  */
 template <typename T>
-CUDA_HOST_DEVICE_CALLABLE bool is_valid(bit_mask_t const * __restrict__ valid, T record_idx) {
+CUDA_HOST_DEVICE_CALLABLE bool is_valid(bit_mask_t const * valid, T bit_index) {
```

Comment:
Hmmm, we may need another accessor that can be `__restrict__`, then?

Reply:
Umm... are you asking a question or making a statement? ... 
anyway, `__restrict__` is only useful when we have at least two reference/pointer parameters and one is writable, which is not the case here IIANM.

Reply:
A bit of both. `const * __restrict__` indicates to the CUDA compiler that it can use the read-only cache (LDG instruction) on architectures that support it. It doesn't require at least two pointer parameters.  My point is that this change could hurt performance.

Reply:
I was thinking of the LDG optimization which the compiler can make for `const * restrict` parameters. However I think it only applies to *kernel* parameters, not device function parameters. Not sure though... We can revisit later.